docs/porting-addons-to-v2.md

```diff
@@ -125,6 +125,14 @@ The steps:
    ember s
    ember test
    ```
+
+1. The lint scripts are expected to work the same way as before inside the test-app. However, there's one common issue you may encounter when running the linter if you use `eslint-plugin-n` or `eslint-plugin-node`: 
+   ```sh
+   error  "@embroider/test-setup" is not published  n/no-unpublished-require
+   error  "ember-cli" is not published  n/no-unpublished-require
+   ```
+   The [lint rule](https://github.com/eslint-community/eslint-plugin-n/blob/master/docs/rules/no-unpublished-require.md) tells that `"@embroider/test-setup"` and `"ember-cli"` are `devDependencies` being imported with `require()`. It's not a problem since the test-app is not meant to be published on npm. One solution is to create `.npmignore` file in the test-app folder and add `ember-cli-build.js` to it (see how [ember-welcome-page ignores all files](https://github.com/ember-cli/ember-welcome-page/blob/main/test-app/.npmignore)). Another solution is to consider the rule should not apply to private packages and deactivate it.
```

Comment:
I don't think we should recommend `.npmignore`, there is just too much to ignore.

If using the `files` entry in package.json, like in the blueprint, the "no-unpublished" rule will be handled for you:
Example: https://github.com/embroider-build/addon-blueprint/blob/main/files/__addonLocation__/package.json#L11
 
this is an _allow list_, which is much shorter than a block list that you'd end up creating with `.npmignore`.

both `no-unpublished-require` and `no-unpublished-import` respect this entry.

in `eslint-plugin-n`, the the rules respect `private` fields, and turn themselves off, I believe. good to call out, good to test to see if we don't need to configure anything as well.

Reply:
> I don't think we should recommend `.npmignore`

I can drop that part 👍

> using the `files` entry

The issue happens in the test-app so I am not sure how to use the `files` field in that case. As I see it, it's really that we shouldn't have this problem with a `private` package, but it turns out we often have it.

> in `eslint-plugin-n`, the rules respect private fields, and turn themselves off, I believe

I did set the `private` field but it had no effect 🤔 I'll double-check once more (my addon initially used `eslint-plugin-node`, so maybe I missed something).

Reply:
Alright, I reworked it a bit, what do you think?